File: scripts/fetch-simple-icons.mjs
/**
 * 从 Simple Icons CDN 拉取图标数据并生成 deploy/cloudflare/assets/simple-icons.json。
 * 运行：node scripts/fetch-simple-icons.mjs
 */

import { writeFileSync } from "fs";
import { resolve, dirname } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const ROOT = resolve(__dirname, "..");
const OUTPUT_PATH = resolve(ROOT, "deploy/cloudflare/assets/simple-icons.json");
const CDN_URL = "https://cdn.simpleicons.org/simple-icons.json";

async function main() {
  console.log(`Fetching Simple Icons from ${CDN_URL} ...`);
  const resp = await fetch(CDN_URL, { signal: AbortSignal.timeout(15000) });
  if (!resp.ok) {
    console.error(`Failed to fetch: ${resp.status} ${resp.statusText}`);
    process.exit(1);
  }
  const data = await resp.json() as Record<string, unknown>;
  console.log(`Fetched ${Object.keys(data).length} icons.`);
  writeFileSync(OUTPUT_PATH, JSON.stringify(data, null, 2), "utf-8");
  console.log(`Written to ${OUTPUT_PATH}`);
}

main().catch((err) => { console.error(err); process.exit(1); });

File: scripts/fetch-simple-icons.py
#!/usr/bin/env python3
"""拉取 Simple Icons 完整数据（含 slug），写入共享数据源 + Cloudflare assets。

共享数据源：仓库根目录 simple-icons.json（Python 端启动时读取）。
Cloudflare 副本：deploy/cloudflare/assets/simple-icons.json（Workers Assets 服务）。
"""
import json
import urllib.request
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]

SRC = "https://unpkg.com/simple-icons@latest/icons.json"
DST_SHARED = ROOT / "simple-icons.json"
DST_CF = ROOT / "deploy" / "cloudflare" / "assets" / "simple-icons.json"

req = urllib.request.Request(SRC, headers={"User-Agent": "PrivLink/1.0"})
with urllib.request.urlopen(req, timeout=20) as resp:
    data = json.loads(resp.read())

print(f"Total icons: {len(data)}")

for dst in (DST_SHARED, DST_CF):
    dst.parent.mkdir(parents=True, exist_ok=True)
    with open(dst, "w", encoding="utf-8") as f:
        json.dump(data, f, ensure_ascii=False, indent=2)
    print(f"Written to {dst}")